From e6727bcfaf6d2cab9dbd3f8012616cad74aee939 Mon Sep 17 00:00:00 2001
From: Marcin Wojtas <mw@semihalf.com>
Date: Thu, 25 Aug 2016 21:07:41 +0200
Subject: [PATCH 0917/2241] clk: mvebu: add CLK_IS_ROOT to CP110_CORE_APLL
 clock

CLK_IS_ROOT clock for the tree source clock will be deprecated
in next LK revisions. For now it has to be restored, in order
to be able to calculate rate properly.

This patch is part of patch-set which adds
mainline mvebu clock gating support [13/21].

Change-Id: I8e39fbac56dec519764fc1574d0f801ba8665995
Signed-off-by: Marcin Wojtas <mw@semihalf.com>
Signed-off-by: Igal Liberman <igall@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/32398
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
Tested-by: Omri Itach <omrii@marvell.com>
---
 drivers/clk/mvebu/cp110-system-controller.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/clk/mvebu/cp110-system-controller.c b/drivers/clk/mvebu/cp110-system-controller.c
index d22f192..59071c9 100644
--- a/drivers/clk/mvebu/cp110-system-controller.c
+++ b/drivers/clk/mvebu/cp110-system-controller.c
@@ -36,6 +36,7 @@
 #include <linux/platform_device.h>
 #include <linux/regmap.h>
 #include <linux/slab.h>
+#include <linux/clk.h>
 
 #define CP110_PM_CLOCK_GATING_REG	0x220
 #define CP110_NAND_FLASH_CLK_CTRL_REG	0x700
@@ -221,7 +222,7 @@ static int cp110_syscon_clk_probe(struct platform_device *pdev)
 	/* Register the APLL which is the root of the clk tree */
 	of_property_read_string_index(np, "core-clock-output-names",
 				      CP110_CORE_APLL, &apll_name);
-	clk = clk_register_fixed_rate(NULL, apll_name, NULL, 0,
+	clk = clk_register_fixed_rate(NULL, apll_name, NULL, CLK_IS_ROOT,
 				      1000 * 1000 * 1000);
 	if (IS_ERR(clk)) {
 		ret = PTR_ERR(clk);
-- 
2.7.4

